### DISCUSSION TITLE
Solved: USB Library removal

### DISCUSSION THREAD

**Discussion by phreatic62** [rern/rAudio#532] [Feature Requests]
Solved.

HI, Apologies, I'm new to all this, but I've had a search and can;t find an answer.
When I plug in a USB source, I was able to find that I had to refresh the library, but is there any way to flush things from the USB part of the library on removal, either automagically or manually?

Many thanks

**Comment by EarsofTin** [comment]
This works for my NAS files.  Click on folder or subfolder icon that you wish to update then click "update database"

**Reply by phreatic62** [reply]
Thank you, that was it. I thought a library update rescanned everything looking for changes and didn't realise you needed (or could) to direct it to a particular source. This has deleted everything from the USB which is now unplugged.

**Reply by githans1** [reply]
My apologies for barging in, but I may have a similar problem in the sense that my usb flash drive is not detected at all. when I do lsblk only the boot and / partitions on the sd card show up. 

I am running rAudio 1 on a RPI3 with DAC+.

When I click on the usb icon on the library page, I get the message "database not available for this mode" (the same happens when I click on the sd icon).

I see in your text that you can actually get into your usb via that icon. Do I have to set a mode somewhere to be able to go into the attached usb drive an d ply the music files there?

**Reply by rern** [reply]
@githans1,
Try:
- Copying a single `*.mp3` or `*.flac` music file to a USB flash drive.
- If the Library update was not run on inserting the drive, Settings > Library > click an icon on the right.

**Reply by phreatic62** [reply]
@githans1 mine seemed to be detecting addition and removal of the USB, just not updating the library. It was a minor inconvenience (for me anyway) as I keep my music library on a NAS.

**Reply by githans1** [reply]
I did what you suggested: one single *.mp3 file on a usb flash drive (formatted ext4). inserted the drive into my rpi3 (while running rAudio-1): same problem. clicking on the little icon on the right of Settings>Library gives the message "library update done". Clicking after that on the usb icon however still produces the message "database not available for this mode". The flash drive is simply not detected:
[console.txt](https://github.com/rern/rAudio-1/files/7709896/console.txt)

**Reply by rern** [reply]
I think this is a hardware problem, based on output in the attached file.
Try the other 3 ports. If they're just the same, try cleaning them with contact cleaner.

**Reply by githans1** [reply]
Turns out you are right. Thanks for the assistance. I slipped my raudio sd card into the rpi3 I use to run my tv (normally with an sd card running kodi) and there the usb drive was recognized. Kind of strange that all of my usb ports are broken at the same time. could my dac+ be doing that?
